src/js/containers/patch_app.js

```diff
@@ -1,31 +1,131 @@
 import React, { PropTypes } from 'react'
 import RangeMapper from '../util/range_mapper'
-import MIDIRangeSelector from '../components/midi_range_selector'
+import RangeSelector from '../components/range_selector'
 import { connect } from 'react-redux'
+import OptionsSelector from '../components/options_selector'
 
 const rangeMapper = RangeMapper(0, 127, 0, 10)
 
+function capitalizeFirstLetter(string) {
+  return string.charAt(0).toUpperCase() + string.slice(1);
+}
+
+function extractRangeValue(event) {
+  return rangeMapper.unmap(parseInt(event.target.value, 10))
+}
+
+function extractBooleanValue(event) {
+  return event.target.value === 'true'
+}
+
 const PatchApp = (props) => {
-  function setParam(path) {
+  function valueAt(path) {
+    return props.patchState.getIn(path.split('.'))
+  }
+
+  function setParam(path, valueExtractor) {
     return (event) => {
       props.dispatch({
         type: 'SET_PARAM',
         path: path,
-        value: rangeMapper.unmap(parseInt(event.target.value, 10))
+        value: valueExtractor(event)
       })
     }
   }
 
   function rangeSelectorFor(path, name) {
-    const value = props.patchState.getIn(path.split('.'))
+    const value = valueAt(path)
     const displayValue = rangeMapper.map(value)
 
     return (
-      <MIDIRangeSelector
+      <RangeSelector
         name={ name }
         path={ path }
         value={ displayValue }
-        onChange={ setParam(path) }
+        onChange={ setParam(path, extractRangeValue) }
+      />
+    )
+  }
+
+  function booleanSelectorFor(path, name, trueText = 'on', falseText = 'off') {
+    const boolVal = valueAt(path)
+    const options = [
+      {
+        optionName: trueText,
+        value: true,
+        checked: boolVal
+      },
+      {
+        optionName: falseText,
+        value: false,
+        checked: !boolVal
+      }
+    ]
+
+    return (
+      <OptionsSelector
+        name={ name }
+        onChange={ setParam(path, extractBooleanValue) }
+        options={ options }
+      />
+    )
+  }
+
+  function enumSelectorFor(path, name, values) {
+    const options = values.map(value => {
+      return {
+        optionName: capitalizeFirstLetter(value),
+        value: value,
+        checked: valueAt(path) === value
+      }
+    })
+
+    return (
+      <OptionsSelector
+        name={ name }
+        onChange={ setParam(path, event => event.target.value) }
+        options={ options }
+      />
+    )
+  }
+
+  function juno106WaveFormSelector() {
```

Comment:
@dbalatero this wouldn't need to exist and could be replaced by a call to `enumSelectorFor` if the midi-synth-mapper lib presented the waveform with a single value and interpreted it back into two bits internally.